include/boost/asio/detail/config.hpp

```diff
@@ -769,14 +769,21 @@
 #endif // !defined(BOOST_ASIO_HAS_STD_FUTURE)
 
 // Standard library support for experimental::string_view.
+// Note: libc++ 7.0 keeps <experimental/string_view> header but deprecates it
+// with an #error directive.
 #if !defined(BOOST_ASIO_HAS_STD_STRING_VIEW)
 # if !defined(BOOST_ASIO_DISABLE_STD_STRING_VIEW)
 #  if defined(__clang__)
 #   if (__cplusplus >= 201402)
 #    if __has_include(<experimental/string_view>)
-#     define BOOST_ASIO_HAS_STD_STRING_VIEW 1
-#     define BOOST_ASIO_HAS_STD_EXPERIMENTAL_STRING_VIEW 1
+#     if defined(_LIBCPP_LFTS_STRING_VIEW)
```

Comment:
`_LIBCPP_LFTS_STRING_VIEW` is the guard defined by `#include <experimental/string_view>` but `#include` cannot be used without bumping into `#error` on libc++ >= 7.